tuf/signed/errors.go

```diff
@@ -15,11 +15,17 @@ type ErrInsufficientSignatures struct {
 
 func (e ErrInsufficientSignatures) Error() string {
 	candidates := strings.Join(e.MissingKeyIDs, ", ")
-	if e.FoundKeys == 0 {
-		return fmt.Sprintf("signing keys not available, need %d keys out of: %s", e.NeededKeys, candidates)
+	switch {
+	case len(e.MissingKeyIDs) < e.NeededKeys:
+		return fmt.Sprintf(
+			"cannot sign because while %d signatures are needed, an insufficient number of valid signing keys have been specified",
```

Comment:
As per discussion IRL, this is basically just a role validity check.  The only way we can get to this weird invalid state is with Witness, so we should just move this check to that functionality.